src/agent/onefuzz-file-format/tests/integration.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use pretty_assertions::assert_eq;
+
+use anyhow::Result;
+use coverage::binary::{BinaryCoverage, Count, FilePath, ModuleBinaryCoverage, Offset};
+use onefuzz_file_format::coverage::binary::{v0, v1};
+
+fn expected_binary_coverage() -> Result<BinaryCoverage> {
+    let main_exe_path = FilePath::new("/setup/main.exe")?;
+    let some_dll_path = FilePath::new("/setup/lib/some.dll")?;
+
+    let mut main_exe = ModuleBinaryCoverage::default();
+    main_exe.offsets.insert(Offset(1), Count(0));
+    main_exe.offsets.insert(Offset(300), Count(1));
+    main_exe.offsets.insert(Offset(5000), Count(0));
+
+    let mut some_dll = ModuleBinaryCoverage::default();
+    some_dll.offsets.insert(Offset(123), Count(0));
+    some_dll.offsets.insert(Offset(456), Count(10));
+
+    let mut binary = BinaryCoverage::default();
+    binary.modules.insert(some_dll_path, some_dll);
+    binary.modules.insert(main_exe_path, main_exe);
+
+    Ok(binary)
+}
+
+#[test]
+fn test_binary_coverage_formats() -> Result<()> {
```

Comment:
This would be a good opportunity to use golden-file testing (e.g. `goldenfile`), but that can be done at a later stage.